### DISCUSSION TITLE
Does argo rollouts provide a Web UI ?

### DISCUSSION THREAD

**Discussion by alwaysastudent** [argoproj/argo-rollouts#1207] [Q&A] [answered]
We use argocd as our GitOps deployment manager, and argocd comes with a Web UI that provides a real-time view of deployment activity and application state in our K8s cluster. 

Can argo-rollouts be integrated with the argocd remote Web UI (which is already integrated with SSO), so that we can use the same UI to perform rollout functions as well - (pause/promote/abort/retry)?

**Comment by jessesuen** [comment, marked as answer]
Not yet, but we're actively thinking about this. See: 
* https://github.com/argoproj/argo-cd/pull/6240
* https://github.com/argoproj/argo-cd/pull/6232

**Reply by alwaysastudent** [reply]
Thank you for the answer. Awesome work you guys have been doing.

Web UI integration would make this project much more complete. 

Also looking forward to the support for multiple kinds of workloads including DS, STS, Jobs/Cronjobs

**Comment by alwaysastudent** [comment]
@jessesuen has there been any progress on this? it seems like the feature was not merged in ArgoCD project

**Reply by jessesuen** [reply]
> Can argo-rollouts be integrated with the argocd remote Web UI (which is already integrated with SSO), so that we can use the same UI to perform rollout functions as well - (pause/promote/abort/retry)?

I should have explained earlier that the ability to perform actions on a Rollout is already integrated with Argo CD. We provide Rollout specific actions from the UI and CLI. This should be working today:
![image](https://user-images.githubusercontent.com/12677113/136465394-1935ff04-0077-40bb-bb75-7a3fc4ad3b49.png)

What is coming is UI integration of the dashboard through an Argo CD extension mechanism. You can see this in action here:
https://cd.apps.argoproj.io/applications/rollouts-demo-istio?resource=&node=argoproj.io%2FRollout%2Frollouts-demo-istio%2Fistio-rollout%2F0&tab=extension
![image](https://user-images.githubusercontent.com/12677113/136465198-c0ef6bfe-5d00-4873-a5f2-1ef2a8d8cbdc.png)

This is still actively being worked on, and is the subject of our upcoming KubeCon talk:
https://kccncna2021.sched.com/event/lV89/the-argo-ecosystem-tailoring-your-installation-through-community-add-ons-jesse-suen-akuity-alexander-matyushentsev-intuit?iframe=no&w=100%&sidebar=yes&bg=no

**Reply by daydy16** [reply]
Hey, is there any news on this? We would love to see the rollout dashboard integrated in argo's ui
